src/MultiRegion/cubed_sphere_boundary_conditions.jl

```diff
@@ -0,0 +1,352 @@
+    grid = field.grid
+
+    Nx, Ny, Nz = size(grid)
+    Hx, Hy, Hz = halo_size(grid)
+
+    Nx == Ny || error("horizontal grid size Nx and Ny must be the same")
+    Nc = Nx
+
+    Hx == Hy || error("horizontal halo size Hx and Hy must be the same")
+    Hc = Hx
+
+    #-- one pass: only use interior-point values:
+    for region in 1:6
+
+        region_E, region_N, region_W, region_S = find_neighboring_panels(region)
+
+        if mod(region, 2) == 1
+            #- odd face number (1, 3, 5):
+            for k in -Hz+1:Nz+Hz
+                #- E + W Halo for field:
+                field[region][Nc+1:Nc+Hc, 1:Nc, k] .=         field[region_E][1:Hc, 1:Nc, k]
+                field[region][1-Hc:0, 1:Nc, k]     .= reverse(field[region_W][1:Nc, Nc+1-Hc:Nc, k], dims=1)'
+                #- N + S Halo for field:
+                field[region][1:Nc, Nc+1:Nc+Hc, k] .= reverse(field[region_N][1:Hc, 1:Nc, k], dims=2)'
+                field[region][1:Nc, 1-Hc:0, k]     .=         field[region_S][1:Nc, Nc+1-Hc:Nc, k]
+            end
+        else
+            #- even face number (2, 4, 6):
+            for k in -Hz+1:Nz+Hz
+                #- E + W Halo for field:
+                field[region][Nc+1:Nc+Hc, 1:Nc, k] .= reverse(field[region_E][1:Nc, 1:Hc, k], dims=1)'
+                field[region][1-Hc:0, 1:Nc, k]     .=         field[region_W][Nc+1-Hc:Nc, 1:Nc, k]
+                #- N + S Halo for field:
+                field[region][1:Nc, Nc+1:Nc+Hc, k] .=         field[region_N][1:Nc, 1:Hc, k]
+                field[region][1:Nc, 1-Hc:0, k]     .= reverse(field[region_S][Nc+1-Hc:Nc, 1:Nc, k], dims=2)'
+            end
+        end
+    end
+
+    return nothing
+end
+
+function fill_cubed_sphere_halo_regions!(field::CubedSphereField{<:Face, <:Face})
+    grid = field.grid
+
+    Nx, Ny, Nz = size(grid)
+    Hx, Hy, Hz = halo_size(grid)
+
+    Nx == Ny || error("horizontal grid size Nx and Ny must be the same")
+    Nc = Nx
+
+    Hx == Hy || error("horizontal halo size Hx and Hy must be the same")
+    Hc = Hx
+
+    #-- one pass: only use interior-point values:
+    for region in 1:6
+
+        region_E, region_N, region_W, region_S = find_neighboring_panels(region)
+
+        if mod(region, 2) == 1
+            #- odd face number (1, 3, 5):
+            for k in -Hz+1:Nz+Hz
+                #- E + W Halo for field:
+                field[region][Nc+1:Nc+Hc, 1:Nc, k]   .=         field[region_E][1:Hc, 1:Nc, k]
+                field[region][1-Hc:0, 2:Nc+1, k]     .= reverse(field[region_W][1:Nc, Nc+1-Hc:Nc, k], dims=1)'
+                field[region][1-Hc:0, 1, k]          .=         field[region_S][1, Nc+1-Hc:Nc, k]
+                #- N + S Halo for field:
+                field[region][2:Nc+1, Nc+1:Nc+Hc, k] .= reverse(field[region_N][1:Hc, 1:Nc, k], dims=2)'
+                if Hc > 1
+                    field[region][1, Nc+2:Nc+Hc, k]   = reverse(field[region_W][1, Nc+2-Hc:Nc, k])
+                end
```

Comment:
no `.=` here? just `=`? why this difference? a comment is appropriate

Reply:
After some investigation, I have come to understand that the broadcasting assignment operator throws an error when applied to a 1D array (extracted from a cubed sphere field), yet it functions without issue for a 2D array (extracted from a cubed sphere field). The root cause of this behavior remains unclear to me, so this is the temporary workaround we can employ in the meantime.

Reply:
isn't it because of https://github.com/CliMA/Oceananigans.jl/issues/3361 ?

Reply:
#3361 does solve part of the problem. For instance,
```
field[region][Nc+1, Nc+1:Nc+Hc, k]  .= view(field[region_E],  1, 1:Hc, k)
```
works. But
```
field[region][1, Nc+2:Nc+Hc, k] .= reverse(view(field[region_W], 1, Nc+2-Hc:Nc, k)
```
doesn't. I couldn't figure out a way to make `view` work with `reverse`. However, I have updated the assignment statements without employing `reverse` in commit [1663528](https://github.com/CliMA/Oceananigans.jl/pull/3488/commits/166352803a511188a09575ca903f0a62eaf5fff1).

Reply:
make a MWE and put in the issue?

Reply:
Done! 🙂